aws/create_aws_network_resources.py

```diff
@@ -0,0 +1,398 @@
+    AWS_REGION: Specifies the AWS Region to send the request to
+
+Usage:
+Import this module to create AWS network resources and Security Group with Inbound and Outbound rules
+
+    from create_aws_network_resources import create_network_resources, create_security_group, ec2_add_security_rules
+
+    resources = create_aws_network_resources(cidr_block="10,0,0,0/16", site_name="demo", subnet_count=3)
+    security_grp_id = create_security_group(vpc_id=resources['vpc_id'], sec_grp_name="security-demo")
+    ec2_add_security_rules(security_grp_id=security_grp_id, rules_json_file="security_rules_example.json")
+
+
+"""
+import json
+import os
+import sys
+from ipaddress import AddressValueError, IPv4Network
+
+import boto3
+from aws_utils import verify_aws_env_requirements_met
+from botocore.exceptions import ClientError
+from ocp_utilities.logger import get_logger
+
+LOGGER = get_logger(__name__)
+ec2_client = boto3.client("ec2")
+
+
+def ec2_assign_name_tag(res_id, tag_name) -> None:
+    """
+    Assign 'Name' tag value to EC2 resource
+
+    Args:
+        res_id (str): The ID of resource(which can be of 'vpc', 'subnet', etc)
+        tag_name (str): The value of 'Name' tag
+
+    Returns: None
+    """
+    ec2_client.create_tags(
+        Resources=[
+            res_id,
+        ],
+        Tags=[
+            {
+                "Key": "Name",
+                "Value": tag_name,
+            },
+        ],
+    )
+
+
+def get_subnets_cidr(cidr_block, subnet_count) -> list:
+    """
+    Calculates Subnet CIDR address blocks within given VPC CIDR block
+
+    Args:
+        cidr_block (str): The IPv4 network of VPC
+        subnet_count (str): The number of additional bits with which to extend the prefix
+
+    Returns:
+        list: The list of subnet CIDR address blocks
+
+    Raises:
+        AddressValueError: If cidr_block provided in invalid for IPv4 address
+    """
+    subnet_range = []
+    try:
+        net = IPv4Network(cidr_block)
+        prefix = net.prefixlen + (subnet_count * 2)
+        for subnet in net.subnets(new_prefix=prefix):
+            subnet_range.append(str(subnet))
+        return subnet_range
+    except AddressValueError:
+        LOGGER.error(f"Address is invalid for IPv4: {cidr_block}")
+        raise
+
+
+def get_availability_zones() -> list:
+    """
+    Provides a list of zone IDs in availability zone of current AWS Region
+
+    Returns:
+        list: The list of Zone IDs
+
+    Raises:
+        ClientError: If error response is provided by an AWS service to Boto3 client’s request.
+    """
+    zones = []
+    try:
+        av_zones = ec2_client.describe_availability_zones()
+        for zone in av_zones.get("AvailabilityZones"):
+            zones.append(zone.get("ZoneId"))
+        return zones
+    except ClientError as ex:
+        LOGGER.error(ex)
+        raise
+
+
+def create_network_resources(cidr_block, site_name, subnet_count) -> dict:
```

Comment:
Not a full review, please break this function. it's too long.

Reply:
ack